src/Compilers/Core/Portable/Text/TextLineCollection.cs

```diff
@@ -62,6 +62,11 @@ public LinePositionSpan GetLinePositionSpan(TextSpan span)
         /// </summary>
         public int GetPosition(LinePosition position)
         {
+            if (position.Line >= this.Count)
+            {
+                throw new ArgumentOutOfRangeException(nameof(position.Line), string.Format(CodeAnalysisResources.LineCannotBeGreaterThanEnd, position.Line, this.Count));
```

Comment:
>throw new ArgumentOutOfRangeException(nameof(position.Line), string.Format(CodeAnalysisResources.LineCannotBeGreaterThanEnd, position.Line, this.Count)) [](http://example.com/codeflow?start=16&length=152)

Is the resource string necessary? It looks like the caller in `ProtocolConversions` will catch the exception and wrap the exception in an exception that includes the line count.